src/devices/Arduino/ArduinoPwmChannel.cs

```diff
@@ -0,0 +1,128 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Device.Pwm;
+using System.Linq;
+using System.Text;
+
+namespace Iot.Device.Arduino
+{
+    internal class ArduinoPwmChannel : PwmChannel
+    {
+        private readonly ArduinoBoard _board;
+        // The digital pin used
+        private readonly int _pin;
+        private double _dutyCycle;
+        private int _frequency;
+        private bool _enabled;
+
+        /// <summary>
+        /// Create a PWM Channel on the Arduino. Depending on the board, it has 4-8 pins that support PWM.
+        /// This expects to take the normal pin number (i.e. D6, D9) as input
+        /// </summary>
+        /// <param name="board">Reference to Board</param>
+        /// <param name="chip">Always needs to be 0</param>
+        /// <param name="channel">See above. Valid values depend on the board.</param>
+        /// <param name="frequency">This parameter is ignored and exists only for compatibility</param>
+        /// <param name="dutyCyclePercentage">PWM duty cycle (0 - 1)</param>
+        public ArduinoPwmChannel(ArduinoBoard board,
+            int chip,
+            int channel,
+            int frequency = 400,
+            double dutyCyclePercentage = 0.5)
+        {
+            Chip = chip;
+            if (chip != 0)
+            {
+                throw new NotSupportedException("No such chip: {0}");
+            }
+
+            Channel = channel;
+            _pin = channel;
+            var caps = board.SupportedPinConfigurations.FirstOrDefault(x => x.Pin == _pin);
```

Comment:
should you check for both mode and pin number here? is it possible that you find something with matching pin but different mode so the below check fails but there will be another pin which matches this?

Reply:
No. The `SupportedPinConfigurations` structure contains the possible modes for each pin. Here, we get the set of modes that particular `_pin` supports. If it does not support PWM, we throw.